packages/itwin/measure-tools/src/tools/MeasureDistanceTool.ts

```diff
@@ -51,19 +57,25 @@ MeasureDistanceToolModel
     return MeasureToolsFeatures.Tools_MeasureDistance;
   }
 
-  constructor() {
+  constructor(enableSheetMeasurements = false) {
     super();
+    this._enableSheetMeasurements = enableSheetMeasurements;
   }
 
   public async onRestartTool(): Promise<void> {
-    const tool = new MeasureDistanceTool();
+    const tool = new MeasureDistanceTool(this._enableSheetMeasurements);
     if (await tool.run()) return;
 
     return this.exitTool();
   }
 
+  private resetSheetData(): void {
```

Comment:
Shouldn't `this.toolModel.reset(false)` take care of this?
(it's already called by super.onReinitialize)